lib/livebook_web/plugs/user_plug.ex

```diff
@@ -39,17 +40,26 @@ defmodule LivebookWeb.UserPlug do
     end
   end
 
+  defp ensure_user_data(%{req_cookies: %{"lb:user_data" => _}} = conn), do: conn
```

Comment:
I believe this is wrong, we always want to run the authentication code, even if the cookie is set. I am now trying to understand if we should always write the cookie as well.

Reply:
Ok, I know what to do. We need to break this apart:

1. The first part is to get the identity data by calling `module.authenticate(...)`. If data is returned, it must be written to the session, otherwise forbidden

2. Then we proceed to get or set the lb:user_data, exactly how it was done before, and copy it to the session (the same as before). The difference is, if "lb:user_data" is not set, the initial value will come from the "identity_data".

3. Then, in the LiveView, we must merge the identity data on top of user data here: https://github.com/livebook-dev/livebook/blob/main/lib/livebook_web/live/hooks/user_hook.ex#L40